src/core/main.c

```diff
@@ -1546,6 +1546,11 @@ static int fixup_environment(void) {
         if (setenv("TERM", t, 1) < 0)
                 return -errno;
 
+        /* The kernels sets HOME=/ for init. Let's undo this. */
+        if (streq_ptr(getenv("HOME"), "/") &&
+            setenv("HOME", "/root", 1) < 0)
+                log_warning_errno(errno, "Failed to set $HOME: %m");
```

Comment:
i must say, I'd probably just unset the whole env var if we are PID 1

Reply:
Yeah, I wasn't sure what the best option is. Let's unset it.